OptimizelySDK/Bucketing/DecisionService.cs

```diff
@@ -164,30 +167,29 @@ public DecisionService(Bucketer bucketer, IErrorHandler errorHandler, UserProfil
                 var bucketingIdResult = GetBucketingId(userId, filteredAttributes);
                 reasons += bucketingIdResult.DecisionReasons;
 
-                variationResult = Bucketer.Bucket(config, experiment, bucketingIdResult.ResultObject, userId);
-                reasons += variationResult.DecisionReasons;
+                decisionVariation = Bucketer.Bucket(config, experiment, bucketingIdResult.ResultObject, userId);
+                reasons += decisionVariation?.DecisionReasons;
 
-                if (variationResult.ResultObject?.Key != null)
+                if (decisionVariation?.ResultObject?.Key != null)
                 {
                     if (UserProfileService != null && !ignoreUPS)
                     {
                         var bucketerUserProfile = userProfile ?? new UserProfile(userId, new Dictionary<string, Decision>());
-                        SaveVariation(experiment, variationResult.ResultObject, bucketerUserProfile);
-
+                        SaveVariation(experiment, decisionVariation?.ResultObject, bucketerUserProfile);
                     }
                     else
                         Logger.Log(LogLevel.INFO, "This decision will not be saved since the UserProfileService is null.");
                 }
 
-                return variationResult.SetReasons(reasons);
+                return decisionVariation?.SetReasons(reasons);
```

Comment:
If I understand correct, this will discard all reasons if decisionVariation is null. Reasons are most important when the result variation is null.

Reply:
`reasons` is instantiated at the top of this method. All reasons are collected in the `reasons` object, so no `reason` is lost.

Reply:
I see this issue is fixed by removing "?".